File: solutions/3799-unique-3-digit-even-numbers/unique-3-digit-even-numbers.go
func totalNumbers(digits []int) int {
	n := len(digits)
	vis := make([]bool, 1000)
	ans := 0

	for i := 0; i < n; i++ {
		if digits[i] == 0 {
			continue
		}
		for j := 0; j < n; j++ {
			if j == i {
				continue
			}
			for k := 0; k < n; k++ {
				if k == i || k == j || digits[k]%2 != 0 {
					continue
				}
				x := digits[i]*100 + digits[j]*10 + digits[k]
				if !vis[x] {
					vis[x] = true
					ans++
				}
			}
		}
	}

	return ans
}

File: solutions/3799-unique-3-digit-even-numbers/unique-3-digit-even-numbers.rs
impl Solution {
    pub fn total_numbers(digits: Vec<i32>) -> i32 {
        let n = digits.len();
        let mut vis = [false; 1000];
        let mut ans = 0;

        for i in 0..n {
            if digits[i] == 0 {
                continue;
            }
            for j in 0..n {
                if j == i {
                    continue;
                }
                for k in 0..n {
                    if k == i || k == j || digits[k] % 2 != 0 {
                        continue;
                    }
                    let x = (digits[i] * 100 + digits[j] * 10 + digits[k]) as usize;
                    if !vis[x] {
                        vis[x] = true;
                        ans += 1;
                    }
                }
            }
        }

        ans
    }
}